src/components/user.js

```diff
@@ -203,17 +223,15 @@ export default class User extends React.Component {
                         disabled={!isFormFieldsValid}
                         onClick={this._handleClickSubmit}>Submit</button> :
                     (
-                        <button className="ButtonEdit"
-                                disabled={!isFormFieldsValid && editMode}
-                                onClick={editMode ? this._handleClickSave : this._handleClickEdit}>
-                            {editMode ? 'Save' : 'Edit'}
-                        </button>
+                        <Edit disabled={!isFormFieldsValid}
+                              onClickSave={this._handleClickSave}
+                              onClickEdit={this._handleClickEdit}
```

Comment:
all `Edit`'s props for event handling (onClickSave, onClickEdit, onClickUndo) just call `setState`
Maybe we can pass only one prop `onChange` that will get an object and call `setState` for it?